server/controllers/articles.js

```diff
@@ -0,0 +1,91 @@
+import { validationResponse } from '@helpers/validationResponse';
+import Response from '@helpers/Response';
+import models from '@models';
+import validateRating from '@validations/rating';
+
+const { User, Article } = models;
+
+/**
+ * @exports ArticleController
+ * @class ArticleController
+ * @description Handles Article related actions
+ * */
+class ArticleController {
+  /**
+   * Rate an article
+   * @async
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @param {object} next The next middleware
+   * @return {json} Returns json object
+   * @static
+   */
+  static async rate(req, res, next) {
+    const userId = req.decoded.id;
+
+    try {
+      // Validate the rating
+      const articleDetails = await validateRating(req.body);
+
+      const { rate } = articleDetails;
+      const { articleId } = req.params;
+
+      // Check if the article exists
+      const article = await Article.findOne({
+        where: { id: articleId },
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: [
+              'id',
+              'username',
+              'email',
+              'firstname',
+              'lastname',
+              'middlename',
+              'active'
+            ]
+          }
+        ]
+      });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+      let result;
+
+      // Check if there is a rating
+      const userRating = await article.getRating({ where: { userId } });
+
+      const newRating = { userId, ratings: rate };
+
+      let statusCode;
+      if (userRating) {
```

Comment:
please remove all of these

Reply:
I appreciate all the feedback. I will implement them immediately.
Thanks for your support